pallets/basic-token/src/lib.rs

```diff
@@ -1,86 +1,111 @@
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
 
 //! Simple Token Transfer
 //! 1. set total supply
 //! 2. establish ownership upon configuration of circulating tokens
 //! 3. coordinate token transfers with the runtime functions
-use frame_support::{
-	decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, ensure,
-};
-use frame_system::ensure_signed;
+
+pub use pallet::*;
 
 #[cfg(test)]
 mod tests;
 
-pub trait Config: frame_system::Config {
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as Token {
-		pub Balances get(fn get_balance): map hasher(blake2_128_concat) T::AccountId => u64;
-
-		pub TotalSupply get(fn total_supply): u64 = 21000000;
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*};
+	use frame_system::pallet_prelude::*;
 
-		Init get(fn is_init): bool;
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
```

Comment:
Oh wow, I didn't realize there were pallets actually using the old decl_ macros still. Good catch.